### PR TITLE
Add useGoodBye

### LINKED ISSUE #17: Hook? :)
Would be lovely to get hook version of this package. It would make things much cleaner :)

### PR DESCRIPTION
**Pull request by FredyC**
Closes #17 

It seems to be working just fine. Had to reimplement `Prompt` from `react-router` since hooks cannot render, but ultimately it's just a couple lines of code of imperative code.

### CODE REVIEW

**Comment by xJkit** [comment]
Thanks for this PR. Is there any plan to write manual guide on README?

File: src/index.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import GoodByeContext from './GoodByeContext';
 
 export Provider from './GoodByeProvider';
 export withGoodBye from './withGoodBye';
-export useGoodBye from './useGoodBye';
+export useGoodBye from './useGoodbye';
```
Comment (xJkit) [inline src/index.js:9]: Perhaps  `useGoodBye` is better than `useGoodbye` in consistence with the others?

Comment (danielkcz) [reply src/index.js:9]: Well, you do have `withGoodBye` and `GoodByeProvider` just above. So I am consistent ;)

Comment (danielkcz) [reply src/index.js:9]: Oh you mean filename. Sure, if that's only thing that is bothering you ;)

**Comment by danielkcz** [comment]
@xJkit Feel free to take over this PR or ditch it. I no longer use the whole goodbye concept so I am not motivated to finish it up.

**Pull request closed without merging**